### PR TITLE
feat(server):add essential plan

### PR DESCRIPTION
**Pull request by yuval-hazaz**
Close: #8631

## PR Checklist

- [ ] Tests for the changes have been added
- [ ] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/CODE_OF_CONDUCT.md) file for detailed contributing guidelines.

### CODE REVIEW

File: packages/amplication-client/src/Purchase/PurchasePage.tsx [hunk 1]
```diff
@@ -22,6 +23,8 @@ import { PROVISION_SUBSCRIPTION } from "../Workspaces/queries/workspaceQueries";
 import { PurchaseLoader } from "./PurchaseLoader";
 import { FAQ } from "./FAQ";
 import { GET_CONTACT_US_LINK } from "../Workspaces/queries/workspaceQueries";
+import { formatError } from "../util/error";
+import { set } from "lodash";
```
Comment (overbit) [inline packages/amplication-client/src/Purchase/PurchasePage.tsx:27]: This import is not used

File: packages/amplication-client/src/Purchase/PurchasePage.tsx [hunk 2]
```diff
@@ -76,17 +81,34 @@ const PurchasePage = (props) => {
...
+    {
+      loading: provisionSubscriptionLoading,
+      error: provisionSubscriptionError,
+    },
+  ] = useMutation<DType>(PROVISION_SUBSCRIPTION, {
+    onCompleted: (data) => {
+      const { provisionStatus, checkoutUrl } = data.provisionSubscription;
+      if (provisionStatus === "PAYMENT_REQUIRED") {
+        window.location.href = checkoutUrl;
+      } else if (provisionStatus === "SUCCESS") {
```
Comment (overbit) [inline packages/amplication-client/src/Purchase/PurchasePage.tsx:95]: Ideally we should use the Enum for provisionStatus and not an hardcoded string.

Update server to return the correct enum in `ProvisionSubscriptionResult` as follow:

```ts
import { Field, ObjectType, registerEnumType } from "@nestjs/graphql";
import { ProvisionSubscriptionStatus } from "@stigg/node-server-sdk";

@ObjectType({
  isAbstract: true,
})
export class ProvisionSubscriptionResult {
  @Field(() => ProvisionSubscriptionStatus, {
    nullable: false,
  })
  provisionStatus!: ProvisionSubscriptionStatus;

  @Field(() => String, {
    nullable: true,
  })
  checkoutUrl?: string;
}

registerEnumType(ProvisionSubscriptionStatus, {
  name: "ProvisionSubscriptionStatus",
});
```

File: packages/amplication-client/src/Workspaces/hooks/useUpgradeButtonData.ts
Comment (overbit) [inline packages/amplication-client/src/Workspaces/hooks/useUpgradeButtonData.ts:1]: @yuval-hazaz check what these changes will do to the Pro users as they will experience different behaviours

File: packages/amplication-client/src/Purchase/PurchasePage.tsx (hunk 2 above)
Comment (yuval-hazaz) [reply packages/amplication-client/src/Purchase/PurchasePage.tsx:95]: I agree with the need but prefer not to go with this change in the current scope. 
I will keep the existing code as it is, and we can consider making the change in a separate scope after considering the impact and different options

Comment (yuval-hazaz) [reply packages/amplication-client/src/Purchase/PurchasePage.tsx:95]: I opened a new issue
https://github.com/amplication/amplication/issues/8639

**Review by overbit — APPROVED** [review]

**Pull request merged**